### PR TITLE
934 init haproxy

### PR DESCRIPTION
**Pull request by bohendo**
## The Problem

nginx is a subpar proxy, it has trouble keeping the ws connection alive

## The Solution

use a better one: haproxy

## Checklist:

- [ ] I am making this PR against staging, not master
- [ ] My code follows the code style of this project.
- [ ] I have described any backwards-incompatibility implications above.
- [ ] I have highlighted which parts of the code should be reviewed most carefully.
- [ ] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

**Comment by kthomas** [comment]
This PR is a lot bigger than just haproxy 🤣 

That aside, the haproxy bits look good so far. How have you tested this?

File: ops/proxy/indra/prod.cfg [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  option httplog
+  option forwardfor
+  option redispatch
+  timeout connect 3000    # 3 seconds
+  timeout client  300000  # 5 minutes
+  timeout server  300000  # 5 minutes
+
+frontend http_443
+  bind *:443 ssl crt fullchain.pem
+  mode http
```
Comment (kthomas) [inline ops/proxy/indra/prod.cfg]: This should be `tcp` for the NATS websocket frontend.

**Comment by bohendo** [comment]
> How have you tested this?
Normal tests that make sure everything works both in browser (ws) and between two node processes (tcp). No load tests or anything more specific yet (eg not sure how to test keep-alive stuff)

**Comment by bohendo** [comment]
> How have you tested this?
Normal tests that make sure everything works both in browser (ws) and between two node processes (tcp). No load tests or anything more specific yet (eg not sure how to test keep-alive stuff)

Also, what's the difference between connection to the nats ws port 4221 via an http-mode proxy vs a tcp-mode one (isn't ws on layer 7 just like http?). You'll see I'm currently doing both in my indra proxy config

**Pull request merged**